Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/components/ViewSettingsController.java

```diff
@@ -248,10 +263,14 @@ public void initializeEventListeners(Object data) {
         });
 
         defaultCost.addEventListener("onClick", event -> {
-                if (disabled) return;
-                String key = "mean";
-                selectComboboxByKey(costAggSelector, key);
-                selectCostViz();
+            if (disabled || costDisabled) return;
+            if (userOptions.getCostTable().isDefault()) {
+                showNoCostSetDialog();
+                return;
```

Comment:
It is more accurate to check all costs are zero, not default cost because the default cost can be non-zero (it might be changed later). The meaning here is all zeros mean cost table is not set as described in the Jira card.
It could be safer if we implement a CostTable.isNotSet method where it checks for all zeros, and use this method instead of isDefault. Because if somebody changes the default value to non-zero then this implementation won't work correctly any more.